report-ng/app/src/components/timings/session-timings/session-timings.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Testerra
+ *
+ * (C) 2023, Selina Natschke, Telekom MMS GmbH, Deutsche Telekom AG
+ *
+ * Deutsche Telekom AG and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {autoinject} from 'aurelia-framework';
+import {NavigationInstruction, RouteConfig} from "aurelia-router";
+import {AbstractViewModel} from "../../abstract-view-model";
+import {ECharts, EChartsOption} from 'echarts';
+import "./session-timings.scss";
+import {MethodDetails, StatisticsGenerator} from "services/statistics-generator";
+import {StatusConverter} from "../../../services/status-converter";
+import moment from "moment";
+import {MethodType} from "../../../services/report-model/framework_pb";
+import {ExecutionStatistics} from "../../../services/statistic-models";
+
+@autoinject()
+export class SessionTimings extends AbstractViewModel {
+    private static readonly SESSION_COLOR = '#6897EA';
+    private static readonly BASEURL_COLOR = '#75C6CB';
+    private _chart: ECharts;
+    private _option: EChartsOption;
+    private _executionStatistics: ExecutionStatistics;
+    private _methodDetails;
+    private _baseURLData;
+    private _sessionData;
+    private _hasEnded = false;
+    private _testDuration: Duration | null = null;
+    private _bars;
+
+    constructor(
+        private readonly _statusConverter: StatusConverter,
+        private _statisticsGenerator: StatisticsGenerator,
+    ) {
+        super();
+        this._methodDetails = [];
+        this._bars = [];
+    }
+
+    activate(params: any, routeConfig: RouteConfig, navInstruction: NavigationInstruction) {
+        super.activate(params, routeConfig, navInstruction);
+
+        this._statisticsGenerator.getExecutionStatistics().then(executionStatistics => {
+            this._executionStatistics = executionStatistics;
+
+            this._testDuration = {
+                startTime: this._executionStatistics.executionAggregate.executionContext.contextValues.startTime,
+                endTime: this._executionStatistics.executionAggregate.executionContext.contextValues.endTime,
+            }
+
+            executionStatistics.classStatistics
```

Comment:
Writing it as you propose, `classStatistic `in the last line is not available. 
In the beginning I used classes.ts as a guide for my implementation, because the structure there was: https://github.com/telekom/testerra/blob/master/report-ng/app/src/components/classes/classes.ts

```
executionStatistics.classStatistics
                .map(classStatistics => {
                    // ...
                })
                .filter(classStatistic => {
                    // ...
                })
                .forEach(classStatistic => {
                    let methodContexts = classStatistic.methodContexts;
                    // ...
                    let methodDetails = methodContexts.map(methodContext => {
                        return new MethodDetails(methodContext, classStatistic);
                    });
                    // ...
                    methodDetails.forEach(methodDetails => {
                        // ...
                    });
                });
```
So methodDetails is also set for each classStatistics or am I misunderstanding something?

Reply:
Sorry, you are right. I was missing that from my perspective. This is my final proposal:

```typescript
const allMethodDetails = []

executionStatistics.classStatistics.forEach(classStatistic => {
    allMethodDetails.concat(
        classStatistic.methodContexts
            .filter(methodContext => methodContext.methodType == MethodType.TEST_METHOD)
            .map(methodContext => new MethodDetails(methodContext, classStatistic))
    )
})

this._methodDetails = allMethodDetails
```